libs/ui/src/lib/components/login-modal.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+
+import Image from 'next/image';
+import { Modal } from 'antd';
+import { Dispatch, SetStateAction } from 'react';
+import { GithubButton } from './button/github';
+import { GoogleButton } from './button/google';
+
+interface LoginModalProps {
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+interface ImageProps {
+  source: string;
+  altText: string;
+}
```

Comment:
I think this interface should change its name and be in `ui/src/lib/interfaces`

example of `ui/src/lib/interfaces/image.ts`: 
```ts
export interface Image {
  source: string;
  altText: string;
}
```